### PR TITLE
[DO NOT MERGE] CE-1310 PageViews Dashboard

### PR DESCRIPTION
**Pull request by adamkarminski**
**DO NOT MERGE**

See [CE-1310](https://wikia-inc.atlassian.net/browse/CE-1310)

This is a beta version of a Page Views graph dashboard for a wikia.

### CODE REVIEW

File: extensions/wikia/SpecialPageViews/SpecialPageViewsController.class.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+		$endDate = new DateTime( date( 'Y-m-d' ) );
+		$endDate->sub( new DateInterval( 'P1D' ) );
+		$startDate = clone $endDate;
+		$startDate->sub( new DateInterval( 'P1M' ) );
+
+		$oOutput->set( $this->getReport( $startDate, $endDate ) );
+		$sReturnChart = $oOutput->getHTML();
+
+		$oTmpl = new EasyTemplate( dirname( __FILE__ ) . "/templates/" );
```
Comment (lukaszkonieczny) [inline extensions/wikia/SpecialPageViews/SpecialPageViewsController.class.php]: I'm not sure do we really need EasyTemplate here

File: extensions/wikia/SpecialPageViews/SpecialPageViewsController.class.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		$oReport->frequency = SponsorshipDashboardDateProvider::SD_FREQUENCY_DAY;
+		$oReport->tmpSource = $this->getSource( $startDate, $endDate );
+		$oReport->setLastDateUnits( $startDate->diff( $endDate )->days );
+		$oReport->acceptSource();
+		$oReport->setId( 1 );
+		$oReport->lockSources();
+		return $oReport;
+	}
+
+	private function getSource( $startDate, $endDate ) {
```
Comment (lukaszkonieczny) [inline extensions/wikia/SpecialPageViews/SpecialPageViewsController.class.php]: Maybe we can move this method to ```SpecialPageViewsSourceDatabase.class.php```

File: extensions/wikia/SpecialPageViews/js/SpecialPageViews.js [hunk 3]
```diff
@@ -0,0 +1,473 @@
+var SponsorshipDashboard = function(){
```
Comment (lukaszkonieczny) [inline extensions/wikia/SpecialPageViews/js/SpecialPageViews.js]: This file will require some linting and maybe we can make AMD module from this code

File: extensions/wikia/SpecialPageViews/templates/output/chart.tmpl.php [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+<!-- MAIN-PAGE -->
+<div class="sponsorship-dashboard-description">
+	<strong class="bigFont"><?= isset( $title ) ? $title : '' ?></strong>
+	<?= isset( $description ) ? $description : '' ?>
+</div>
+<?php
+if( isset( $showActionsButton ) ){ ?>
+	<div class="sponsorship-dashboard-actions">
+		<?php
+			$dropdown = [
```
Comment (lukaszkonieczny) [inline extensions/wikia/SpecialPageViews/templates/output/chart.tmpl.php]: We can prepare this data in Controller

File: extensions/wikia/SpecialPageViews/templates/output/chart.tmpl.php [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+<?php } ?>
+<div class="sponsorship-dashboard-panel">
+	<div class="sponsorship-dashboard-panel-body">
+		<div id="placeholder<?= $number ?>" class="placeholder" ></div>
+	</div>
+</div>
+<div class="sponsorship-dashboard-panel-header">
+	<div class="datepicker left">
+		<div id="overviewLabel<?= $number ?>" class="overviewLabel"><?= wfMessage( 'sponsorship-dashboard-from-label' )->escaped() ?></div>
+		<select id="sd-year-from" class="sd-datepicker">
```
Comment (lukaszkonieczny) [inline extensions/wikia/SpecialPageViews/templates/output/chart.tmpl.php]: What do you think about using some JS calendar, just to pick the proper date?

File: extensions/wikia/SpecialPageViews/templates/output/chart.tmpl.php [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+	</div></div>
+	<div class="sd-select-deselect-all">
+		<a class="wikia-button secondary" href="#" id="sponsorship-dashboard-select<?= $number ?>"><?= wfMessage('sponsorship-dashboard-select-all')->escaped() ?></a>
+		<a class="wikia-button secondary" href="#" id="sponsorship-dashboard-deselect<?= $number ?>"><?= wfMessage('sponsorship-dashboard-deselect-all')->escaped() ?></a>
+	</div>
+</div>
+<!-- TODO: no excanvas.pack.js !! -->
+<!--[if IE]><script language="javascript" type="text/javascript" src="excanvas.pack.js"></script><![endif]-->
+<? global $wgExtensionsPath ?>
+<script language="javascript" type="text/javascript" src="<?= $wgExtensionsPath ?>/wikia/SpecialPageViews/js/SpecialPageViews.js"></script>
```
Comment (lukaszkonieczny) [inline extensions/wikia/SpecialPageViews/templates/output/chart.tmpl.php]: I will move adding this script to contoller

File: includes/wikia/DefaultSettings.php [hunk 7]
```diff
@@ -614,6 +614,7 @@
 include_once( "$IP/extensions/wikia/Thumbnails/Thumbnails.setup.php" );
 include_once( "$IP/extensions/wikia/InstantGlobals/InstantGlobals.setup.php" );
 include_once( "$IP/extensions/wikia/UserTools/UserTools.setup.php" );
+include_once( "$IP/extensions/wikia/SpecialPageViews/SpecialPageViews.setup.php" );
```
Comment (lgarczewski) [inline includes/wikia/DefaultSettings.php]: Should not be enabled by default -- we should ask Trevor and Kirk what their rollout plans are.

Comment (adamkarminski) [inline includes/wikia/DefaultSettings.php]: :+1:

**Pull request closed without merging**